src/Templates/MethodTemplate.cshtml

```diff
@@ -177,44 +209,71 @@ func (client @(Model.Owner)) @(Model.ResponderMethodName)(resp *http.Response) (
         @(Model.RespondDecorators.EmitAsArguments()))
 
     @(Model.Response(true))
+</text>
+}
     return
 }
 
+@if (Model.IsLongRunningOperation() && Model.IsPageable)
+{
+    var pageType = Model.ReturnType.Body.Cast<CompositeTypeGo>().UnwrapPageType();
+<text>
+@EmptyLine
+func (client @(Model.Owner)) @(Model.ResponderMethodName.ToCamelCase())(resp *http.Response) (result @pageType.ContentType.Name, err error) {
+    @if (Model.ReturnValue().Body.IsStreamType())
+    {
+    @:result.Value = &resp.Body
+    }
+    err = autorest.Respond(
+        @(Model.RespondDecorators.EmitAsArguments()))
+
+    @(Model.Response(true))
+    return
+}
+</text>
+}
+
 @if (Model.IsPageable && !Model.NextAlreadyDefined)
 {
-    var lastResTypyeName = Model.ReturnValue().Body.Name;
-    var preparerName = Model.ReturnValue().Body.Cast<CompositeTypeGo>().PreparerMethodName;
+    var preparerName = Model.ReturnValue().Body.Cast<CompositeTypeGo>().UnwrapPageType().PreparerMethodName;
     if (Model.IsLongRunningOperation())
     {
-        lastResTypyeName = Model.ReturnValue().Body.Cast<FutureTypeGo>().ResultTypeName;
-        preparerName = $"{lastResTypyeName}Preparer";
+        preparerName = $"{Model.LastResultsTypeName().ToCamelCase()}Preparer";
     }
 <text>
 @EmptyLine
 // @(Model.NextMethodName) retrieves the next set of results, if any.
-func (client @(Model.Owner)) @(Model.NextMethodName)(lastResults @lastResTypyeName) (@Model.MethodReturnSignature()) {
+func (client @(Model.Owner)) @(Model.NextMethodName)(lastResults @Model.LastResultsTypeName()) (@Model.NextMethodReturnSignature()) {
     req, err := lastResults.@(preparerName)()
     if err != nil {
-        return result, @(Model.AutorestError("Failure preparing next results request"))
+        return result, @(Model.AutorestError("Failure preparing next results request", null, null, Model.NextMethodName))
     }
     if req == nil {
         return
     }
     @if (Model.IsLongRunningOperation())
     {
-    @:return client.@(Model.SenderMethodName)(req)
+    <text>
+    var resp *http.Response
+    resp, err = autorest.SendWithSender(client, req,
+        autorest.DoRetryForStatusCodes(client.RetryAttempts, client.RetryDuration, autorest.StatusCodesForRetry...))
+    if err != nil {
+        return result, @(Model.AutorestError("Failure sending next results request", "resp", null, Model.NextMethodName))
+    }
+    return client.@(Model.ResponderMethodName.ToCamelCase())(resp)
+    </text>
     }
     else
     {
     <text>
     resp, err := client.@(Model.SenderMethodName)(req)
     if err != nil {
-        @(Model.Response(false))
-        return result, @(Model.AutorestError("Failure sending next results request", "resp"))
+        @(Model.Response(true))
```

Comment:
>true [](start = 25, length = 4)

nit: same here would put an explicit named parameter to be more clear what `true/false` means